ha/setup/ha_setup.py

```diff
@@ -0,0 +1,299 @@
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import sys
+import errno
+import argparse
+import inspect
+import traceback
+import pathlib
+import os
+
+from cortx.utils.conf_store import Conf
+from cortx.utils.log import Log
+from cortx.utils.validator.v_pkg import PkgV
+from cortx.utils.security.cipher import Cipher
+from ha.execute import SimpleCommand
+from ha import const
+
+class Cmd:
+    """
+    Setup Command
+    """
+    _index = "conf"
+
+    def __init__(self, args: dict):
+        """
+        Init method.
+        """
+        self._url = args.config
+        Conf.load(self._index, self._url)
+        self._args = args.args
+        self._execute = SimpleCommand()
+
+    @property
+    def args(self) -> str:
+        return self._args
+
+    @property
+    def url(self) -> str:
+        return self._url
+
+    @staticmethod
+    def usage(prog: str):
+        """
+        Print usage instructions
+        """
+        sys.stderr.write(
+            f"usage: {prog} [-h] <cmd> <--config url> [--plan <TYPE>] <args>...\n"
+            f"where:\n"
+            f"cmd   post_install, config, init, test, upgrade, reset, cleanup, backup, restore\n"
+            f"--config   Config URL\n"
+            f"--plan   Test plan")
+
+    @staticmethod
+    def get_command(desc: str, argv: dict):
+        """
+        Return the Command after parsing the command line.
+        """
+        parser = argparse.ArgumentParser(desc)
+
+        subparsers = parser.add_subparsers()
+        cmds = inspect.getmembers(sys.modules[__name__])
+        cmds = [(x, y) for x, y in cmds
+            if x.endswith("Cmd") and x != "Cmd"]
+        for name, cmd in cmds:
+            cmd.add_args(subparsers, cmd, name)
+        args = parser.parse_args(argv)
+        return args.command(args)
+
+    @staticmethod
+    def add_args(parser: str, cls: str, name: str):
+        """
+        Add Command args for parsing.
+        """
+        parser1 = parser.add_parser(cls.name, help='setup %s' % name)
+        parser1.add_argument('--config', help='Config URL')
+        parser1.add_argument('--plan', nargs="?", default=[], help='Test plan')
+        parser1.add_argument('args', nargs='*', default=[], help='args')
+        parser1.set_defaults(command=cls)
+
+class PostInstallCmd(Cmd):
```

Comment:
In post install we need to create /etc/cortx/ha dir
and move /opt/seagate/..../ha.conf to above dir

Reply:
@ajay-paratmandali I think it should be after rpm post install (Through spec file) and this post install, right?

Reply:
rpm spec is only to install specific stuff

i think it should be part of setup.
as in future we may need to modify and add more to conf file so good to have all in single location
 
also in 
```
    cleanup:
    - cortxha support_bundle create            cmd: ha_setup cleanup
        args: --config <URL>
```
we need to remove such change

Reply:
done